transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/internal/NetworkConnectionInfo.java

```diff
@@ -0,0 +1,214 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.android.datatransport.cct.internal;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class NetworkConnectionInfo {
+
+  public enum NetworkType {
+    MOBILE(0),
+    WIFI(1),
+    MOBILE_MMS(2),
+    MOBILE_SUPL(3),
+    MOBILE_DUN(4),
+    MOBILE_HIPRI(5),
+    WIMAX(6),
+    BLUETOOTH(7),
+    DUMMY(8),
+    ETHERNET(9),
+    MOBILE_FOTA(10),
+    MOBILE_IMS(11),
+    MOBILE_CBS(12),
+    WIFI_P2P(13),
+    MOBILE_IA(14),
+    MOBILE_EMERGENCY(15),
+    PROXY(16),
+    VPN(17),
+
+    NONE(-1);
+
+    private final int value;
+
+    NetworkType(int value) {
+      this.value = value;
+    }
+
+    public int getValue() {
+      return value;
+    }
+
+    @Nullable
+    public static NetworkType forNumber(int value) {
+      switch (value) {
```

Comment:
Can you maintain a Map<Integer, NetworkType> instead? it would make this method O(1) and imo improve readability, wdyt?

Reply:
Done.